### ISSUE TITLE
Access denied error on IE9

### ISSUE THREAD

**Issue by ruchigoyal2005** [mzabriskie/axios#246]
I am getting access denied when using axios get with IE9. is there any workaround for this?

**Comment by mzabriskie** [comment]
Can you provide more info? Full error message, stack trace, etc. Also are you making a request to a different URI than the origin (www.example.com -> www.anothersite.net)?

**Comment by ruchigoyal2005** [comment]
This is the error description:
{
   [functions]: ,
   description: "Access is denied.",
   message: "Access is denied.",
   name: "Error",
   number: -2147024891
}

It is thrown from `request.open` in xhr.js. yes, I am making request from .net to .com

**Comment by mzabriskie** [comment]
Are you sure you have your server set up correctly? The browser makes a preflight request using the `OPTIONS` request method to verify access. The server should respond to this request with the following headers:

```
Access-Control-Allow-Origin: www.example.com
Access-Control-Allow-Credentials: true
```

This will allow cross origin requests from www.example.com to your server. You can also allow access from all origins by using `Access-Control-Allow-Origin: *`.

**Comment by ruchigoyal2005** [comment]
Finally I got this working. There were couple of issues with IE9
1) I was getting above access denied error because locally I was pointing to Api which has https://. but local was http. In IE9 it should be either http to http or https to https. so change the local code to point to http. 
2) XDomainRequest in IE9 doesn't support custom headers. After removing custom headers atleast my api calls were working. 
3) After fixing 1st and 2nd, I was getting promise rejection. After debugging I found that in xhr.js file in onLoad() response.resonseText is undefined. responseText comes as a part of request. so I did send the pull request to fix this issue.
